pkg/controller/secret/secretsync.go

```diff
@@ -0,0 +1,208 @@
+	}
+	return factory.New().WithSync(c.sync).ResyncEvery(resync).WithSyncDegradedOnError(operatorClient).WithInformers(
+		operatorClient.Informer(),
+		secretInformer.Core().V1().Secrets().Informer(),
+		configMapInformer.Core().V1().Secrets().Informer(),
+	).ToController("SecretSync", eventRecorder)
+}
+
+func (c *SecretSyncController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	opSpec, _, _, err := c.operatorClient.GetOperatorState()
+	if err != nil {
+		klog.V(2).ErrorS(err, "Error while getting operator state")
+		return err
+	}
+	if opSpec.ManagementState != operatorv1.Managed {
+		klog.V(2).Info("Operator management state is not managed")
+		return nil
+	}
+
+	cloudSecret, err := c.secretLister.Secrets(util.OperatorNamespace).Get(util.CloudCredentialSecretName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// TODO: report error after some while?
+			klog.V(2).Infof("Waiting for secret %s from %s", util.CloudCredentialSecretName, util.OperatorNamespace)
+			return nil
+		}
+		klog.V(2).ErrorS(err, "Secret listener failed to get secret details")
+		return err
+	}
+
+	cloudConfConfigMap, err := c.configMapLister.ConfigMaps(util.ConfigMapNamespace).Get(util.ConfigMapName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// TODO: report error after some while?
+			klog.V(2).Infof("Waiting for configmap %s from %s", util.CloudCredentialSecretName, util.ConfigMapNamespace)
+			return nil
+		}
+		klog.V(2).ErrorS(err, "Configmap listener failed to get cm details")
+		return err
+	}
+
+	// Get the storage-secret-store secret to be created from ibm-cloud-credential secret and clod-conf configmap
+	driverSecret, err := c.translateSecret(cloudSecret, cloudConfConfigMap)
+	if err != nil {
+		klog.V(2).ErrorS(err, "Error while extracting data from secret/cm")
+		return err
+	}
+	_, _, err = resourceapply.ApplySecret(c.kubeClient.CoreV1(), c.eventRecorder, driverSecret)
+	if err != nil {
+		klog.V(2).ErrorS(err, "Error while creating the secret")
+		return err
+	}
+	klog.V(2).Infof("%s secret created successfully", util.IBMCSIDriverSecretName)
+	return nil
+}
+
+func (c *SecretSyncController) translateSecret(cloudSecret *v1.Secret, cloudConf *v1.ConfigMap) (*v1.Secret, error) {
+	apiKey, ok := cloudSecret.Data[cloudSecretKey]
+	if !ok {
+		return nil, fmt.Errorf("cloud-credential-operator secret %s did not contain key %s", util.CloudCredentialSecretName, cloudSecretKey)
+	}
+	conf, ok := cloudConf.Data[CloudConfigmapKey]
+	if !ok {
+		return nil, fmt.Errorf("cloud-credential-operator configmap %s did not contain key %s", util.ConfigMapName, cloudSecretKey)
+	}
+
+	var re *regexp.Regexp
+	var match []string
+	//// Extracting the region from configmap
+	//re = regexp.MustCompile("region = (.*?)\n")
+	//match = re.FindStringSubmatch(conf)
+	//if len(match) <= 1 {
+	//	return nil, fmt.Errorf("cloud-credential-operator configmap %s did not contain region", util.ConfigMapName)
+	//}
+	//region := match[1]
```

Comment:
Nit: can this be removed, if it's no longer needed?

Reply:
done